### PR TITLE
feat: add FluxRecord.row with response data stored in Array

### PR DESCRIPTION
**Pull request by michaelahojna**
## Proposed Changes

Adding possibility of accessing response data in Array FluxRecord.row.

In case of using pivot on data, where field contains labels that occur by default in the annotated CSV (f.e. "result" or "table"), could be duplicated column names in response. In that case FluxRecord.values (Hash), which can hold only unique keys, doesn't show complete data. This edge case is solved by using FluxRecord.row (Array).

## Checklist

- [x] CHANGELOG.md updated
- [x] Rebased/mergeable
- [x] A test has been added if appropriate
- [x] `rake test` completes successfully
- [x] Commit messages are [conventional](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Sign [CLA](https://www.influxdata.com/legal/cla/) (if not already signed)

### CODE REVIEW

File: influxdb-client.gemspec [hunk 1]
```diff
@@ -24,15 +24,15 @@ require 'influxdb2/client/version'
...
+  spec.email = ['jakub.bednar@gmail.com']
 
-  spec.summary       = 'Ruby library for InfluxDB 2.'
-  spec.description   = 'This is the official Ruby library for InfluxDB 2.'
-  spec.homepage      = 'https://github.com/influxdata/influxdb-client-ruby'
-  spec.license       = 'MIT'
+  spec.summary = 'Ruby library for InfluxDB 2.'
+  spec.description = 'This is the official Ruby library for InfluxDB 2.'
+  spec.homepage = 'https://github.com/influxdata/influxdb-client-ruby'
+  spec.license = 'MIT'
```
Comment (bednar) [inline influxdb-client.gemspec:35]: Please rollback this file changes, it is not necessary for this PR.

**Review by bednar — CHANGES_REQUESTED** [review]
Thanks for your PR 👍 
There are a few requirements that must be be satisfy before we accept the PR:

File: lib/influxdb2/client/flux_csv_parser.rb [hunk 2]
```diff
@@ -187,6 +187,12 @@ def _add_column_names_and_tags(table, csv)
         column.label = csv[i]
         i += 1
       end
+
+      duplicates = table.columns.group_by { :label }.select { |_k, v| v.size > 1 }
+
+      warning = "The response contains columns with duplicated names: #{duplicates.keys.join(', ')}\nYou should use the
+ 'FluxRecord.row to access your data instead of 'FluxRecord.values' hash."
```
Comment (bednar) [inline lib/influxdb2/client/flux_csv_parser.rb:194]: The warning massage is not properly formatted:

image


```suggestion
      warning = "The response contains columns with duplicated names: #{duplicates.keys.join(', ')}.
You should use the 'FluxRecord.row' to access your data instead of 'FluxRecord.values' hash."
```

**Review by bednar — CHANGES_REQUESTED** [review]

**Review by bednar — APPROVED** [review]
LGTM 🌹
